src/main/java/Pawn.java

```diff
@@ -17,31 +18,35 @@ public long valid_moves(long from, int side, LocationBitboard board){
         //white pawn
         if (side == 0) {
             // pawn's 1 tile ahead move, if there's no piece there already
-            if (from & FileAndRank.RANK_1 == 0L) {
+            if ((from & FileAndRank.RANK_1) == 0L) {
```

Comment:
Looks like you fixed the error

bad operand types for binary operator '&'
            if (from & FileAndRank.RANK_1 == 0L) {
                     ^
  first type:  long
  second type: boolean